### DISCUSSION TITLE
Coverage including comments with no prior newline

### DISCUSSION THREAD

**Discussion by kierankasha** [xd009642/tarpaulin#1308] [Q&A]
I noticed a little bug in a small crate (~100 lines) I'm working on. Comments without a newline before them are getting counted towards coverage. Adding a newline gave me +2.88% coverage. This is the command used:
`cargo tarpaulin -v --release --engine ptrace --all-features  --out html`

This is an excerpt from the html report showing with and without the newline:
![image](https://github.com/xd009642/tarpaulin/assets/79289747/0d56cd3d-38e0-469f-b1d0-29b24c639780)
![image](https://github.com/xd009642/tarpaulin/assets/79289747/579bac50-4728-4485-85ce-2abca99a85c4)

Any ideas for how to resolve this without having to add newlines everywhere?

Mentioned in #351

**Comment by xd009642** [comment]
The best way to fix it is in tarpaulin, so I'll have a look at doing this as soon as I'm able (it'll be in the `src/source_analysis/mod.rs` file if you want to have a look before I get to it

**Comment by xd009642** [comment]
So I've got a PR open for this that looks like it solves it in my test case https://github.com/xd009642/tarpaulin/pull/1318 you can try this if you have time. But single line comment filtering was definitely an issue and is now resolved in this so I'm fairly confident it will be solved as a result.
